nii_reader.py

```diff
@@ -1,17 +1,102 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import io
+import os
+import random
+import string
+import tempfile
+import urllib.request
+from typing import Dict, Optional
+
+import nibabel.filebasedimages
+import numpy as np
 from jina import Executor, DocumentArray, requests
+from jina.logging.logger import JinaLogger
+from jina.types.document import _is_datauri
+from monai.config import DtypeLike
 from monai.data.image_reader import NibabelReader
 
 
 class NiiReader(Executor):
-    def __init__(self, *args, **kwargs):
-        super(NiiReader, self).__init__()
-        self.r = NibabelReader()
+    """
+    Load `NIfTI` format images based on `Nibabel` library.
+    """
+
+    def __init__(
+        self,
+        as_closest_canonical: bool = False,
+        nibabel_args: Optional[Dict] = None,
+        dtype: DtypeLike = np.float32,
```

Comment:
I think we should only use primitive datatypes (that can be converted to/form json), otherwise we can not pass this over network with `uses_with`